components/browser/state/src/main/java/mozilla/components/browser/state/state/content/DownloadState.kt

```diff
@@ -40,40 +40,42 @@ data class DownloadState(
     val destinationDirectory: String = Environment.DIRECTORY_DOWNLOADS,
     val referrerUrl: String? = null,
     val skipConfirmation: Boolean = false,
-    val id: Long = Random.nextLong(),
-    val sessionId: String? = null
+    val id: String = UUID.randomUUID().toString(),
+    val sessionId: String? = null,
+    val createdTime: Long = System.currentTimeMillis()
 ) : Parcelable {
     val filePath: String get() =
         Environment.getExternalStoragePublicDirectory(destinationDirectory).path + "/" + fileName
 
     /**
      * Status that represents every state that a download can be in.
      */
-    enum class Status {
+    @Suppress("MagicNumber")
```

Comment:
Couldn't we just persist the String value and rely on `Enum.valueOf`. Then we don't need the `StatusConverter` etc.

Reply:
It could be a bit brittle if somebody renames some `Status`, at least with `ids` is less possible, what do you think?

Reply:
Well, on the flip side, if someone renames / messes up the names we would get an exception/crash. If someone messes up IDs crazy things can start happening. Eitherway, we need data migration if the status fields change. Up to you :). Don't feel strongly, just thought we can simplify.

Reply:
👍

Reply:
For the moment, I prefer to keep it as it's :)